packages/jest-runtime/src/index.ts

```diff
@@ -887,6 +888,42 @@ class Runtime {
       return this._environment.global.process;
     }
 
+    if (moduleName === 'module') {
+      const createRequire = (modulePath: string | URL) => {
+        const filename =
+          typeof modulePath === 'string'
+            ? modulePath
+            : fileURLToPath(modulePath);
+
+        return this._createRequireImplementation({
+          children: [],
+          exports: {},
+          filename,
+          id: filename,
+          loaded: false,
+        });
+      };
+
+      const overriddenModule = {...nativeModule};
+
+      if ('createRequire' in nativeModule) {
+        Object.defineProperty(overriddenModule, 'createRequire', {
+          value: createRequire,
+        });
+      }
+      if ('createRequireFromPath' in nativeModule) {
+        Object.defineProperty(overriddenModule, 'createRequireFromPath', {
+          value: (filename: string) => createRequire(filename),
```

Comment:
we should throw if it's not a string. Node itself just crashes with a TypeError

```sh-session
$ node -p 'require("module").createRequireFromPath(new URL("file:///something"))'
internal/modules/cjs/loader.js:1049
    filename.endsWith('/') || (isWindows && filename.endsWith('\\'));
             ^

TypeError: filename.endsWith is not a function
    at Function.createRequireFromPath (internal/modules/cjs/loader.js:1049:14)
    at [eval]:1:19
    at Script.runInThisContext (vm.js:116:20)
    at Object.runInThisContext (vm.js:306:38)
    at Object.<anonymous> ([eval]-wrapper:9:26)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at evalScript (internal/process/execution.js:80:25)
    at internal/main/eval_string.js:23:3
```

We can throw a cleaner one (hinting to use `createRequire` instead), but we shouldn't silently make a non-string (`URL` specifically) work